src/aosm/azext_aosm/tests/latest/unit_test/test_definiton_folder_builder/test_definition_folder_builder.py

```diff
@@ -0,0 +1,48 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from pathlib import Path
+from unittest import TestCase
+from unittest.mock import MagicMock, patch
+
+from azext_aosm.definition_folder.builder.definition_folder_builder import DefinitionFolderBuilder
+
+
+class TestDefinitionFolderBuilder(TestCase):
+    """Test the definition folder builder."""
+
+    @patch("pathlib.Path.write_text")
+    @patch("pathlib.Path.mkdir")
+    def test_add_elements_and_write(self, mock_mkdir, mock_write_text):

```

Comment:
I think this works, but there's a lot of mocking, is it really necessary?
From a quick look, I reckon you can construct real BaseElementDefinitions and LocalFileBuilders (with dummy paths and contents) fairly easily. Then you could use a temporary directory to write the files to, and check what's in them, rather than mocking pathlib?
(IMO the real behaviour you want to test is that the files are created with the right names and contents, and so it's better to test that directly if you can rather than indirectly by verifying calls to pathlib).

Reply:
I didn't write this test, so leaving it alone. But we chatted about it